lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -838,36 +840,70 @@ class ADContext {
   SILPassManager &getPassManager() const { return passManager; }
   Lowering::TypeConverter &getTypeConverter() { return module.Types; }
 
-  SmallVectorImpl<DifferentiableFunctionInst *> &
-  getDifferentiableFunctionInsts() {
-    return differentiableFunctionInsts;
+  /// Returns true if the `differentiable_function` instruction worklist is
+  /// empty.
+  bool isDifferentiableFunctionInstsWorklistEmpty() const {
+    return differentiableFunctionInsts.empty();
   }
 
-  SmallPtrSetImpl<DifferentiableFunctionInst *> &
-  getProcessedDifferentiableFunctionInsts() {
-    return processedDifferentiableFunctionInsts;
+  /// Pops and returns a `differentiable_function` instruction from the
+  /// worklist. Returns nullptr if the worklist is empty.
+  DifferentiableFunctionInst* popDifferentiableFunctionInstFromWorklist() {
+    if (differentiableFunctionInsts.empty()) return nullptr;
+    return differentiableFunctionInsts.pop_back_val();
   }
 
-  llvm::SmallMapVector<SILDifferentiabilityWitness *, DifferentiationInvoker,
-                       32> &
-  getInvokers() {
+  /// Adds the given `differentiable_function` instruction to the worklist.
+  void addDifferentiableFunctionInst(DifferentiableFunctionInst* dfi) {
```

Comment:
```suggestion
  void addDifferentiableFunctionInstToWorklist(DifferentiableFunctionInst* dfi) {
```
Sorry for the late comment, feel free to address in a follow-up!

Reply:
OK. will address in a follow up.